03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,26 +1,67 @@
 const Task= require('../models/Task')
+const asyncWrapper = require('../middleware/async')
+const {createCustomError} = require('../errors/custom-error')
 
+const getAllTasks = asyncWrapper(async(req,res) => {
+    
+        const tasks = await Task.find({})
+        res.status(200).json({tasks})
+    
 
-const getAllTasks = (req,res) => {
-    res.send('Get all Tasks')
-}
+    
+})
 
-const createTask = async (req,res) => {
+const createTask = asyncWrapper(async (req,res) => {
+    
     const task= await Task.create(req.body)
     res.status(201).json({task})
-}
+    
+    
+})
 
-const getTask = (req,res) => {
-    res.json({id: req.params.id})
-}
+const getTask = asyncWrapper(async (req,res,next) => {
+   
+        const {id:taskID} =req.params
+        const task = await Task.findOne({_id: taskID })
 
-const updateTask = (req,res) => {
-    res.send('Update Tasks')
-}
+        if(!task)
+        {
+            
+            return next(createCustomError(`No task with id: ${taskID}`,404))
+            
+        }
+      res.status(200) .json({task})
+      
+})
 
-const deleteTask = (req,res) => {
-    res.send('Delete Tasks')
-}
+const updateTask = asyncWrapper(async (req,res) => {
+    
+        const {id:taskID} =req.params
+        const task = await Task.findOneAndUpdate({_id: taskID }, req.body,{
+            new:true,
+            runValidators:true,
+        })
+
+        if(!task)
+        {
+            return next(createCustomError(`No task with id: ${taskID}`,404))
+        }
+      res.status(200) .json({task})
+      
+})
+
+const deleteTask = asyncWrapper(async (req,res) => {
```

Comment:
Add the next param to the params on the async function. You are calling it [here](https://github.com/faridahyderi/node-express-course/blob/2ec25ffc93d1fe6f480ff14378fec87bcabe7b5c/03-task-manager/starter/controllers/tasks.js#L47C1-L48C1) and [here](https://github.com/faridahyderi/node-express-course/blob/2ec25ffc93d1fe6f480ff14378fec87bcabe7b5c/03-task-manager/starter/controllers/tasks.js#L60C1-L61C1)